libs/utils/utils.js

```diff
@@ -649,9 +649,17 @@ export function convertStageLinks({ anchors, config, hostname }) {
     const matchedDomain = Object.keys(domainsMap)
       .find((domain) => a.href.includes(domain));
     if (!matchedDomain) return;
-    a.href = a.href.replace(a.hostname, domainsMap[matchedDomain] === 'origin'
-      ? hostname
-      : domainsMap[matchedDomain]);
+    const convTargetDomain = domainsMap[matchedDomain];
+    a.href = a.href.replace(a.hostname, convTargetDomain === 'origin'
+      ? hostname.to || hostname
+      : convTargetDomain.to || convTargetDomain);
+    if (convTargetDomain.useExt) return;
+    const urlObject = new URL(a.href);
+    const path = urlObject.pathname;
+    const extension = path.split('.').pop();
+    if (!extension) return;
+    urlObject.pathname = path.replace(`.${extension}`, '');
+    a.href = urlObject.toString();
```

Comment:
This looks like added complexity that overlaps with `appendHtmlToLink`. Could we not convert stage links first and then let the logic run as before? That might also mean that we don't need to update the object with the domain maps on the consumer side.

Or are you trying to tackle the use-case where a link has been already authored with `.html`? In that case, wouldn't checking whether the page is a .page/.live one suffice as a check?

Reply:
I'm trying to tackle the use-case where a link has been already authored with `.html`.
I like the .page/.live check approach, this will also remove the need for `stageDomainsMap` updates.
I will steal that 😉

Reply:
Here's some more context to links being authored with [.html](https://adobe-mwp.slack.com/archives/C03B0BUA75E/p1727826864607949) on slack